### ISSUE TITLE
Erroneous double entries when adding directory to playlist

### ISSUE THREAD

**Issue by worstje** [clsid2/mpc-hc#1063]
I observed some strange behaviour today while queueing up some files I freshly organized. I use the MPC-HC bundled with K-Lite Codec pack; detailed version information can be found at the bottom of the issue.

# Observed behaviour

For example, take this file:  "M:\t.A.T.u\[2000] Я сошла с ума\01 - Я Сошла С Ума (Оригинал).flac".

I observe the following behaviours when adding it to my playlist:

✅ **DRAG file "01 - Я Сошла С Ума (Оригинал).flac"**

One file appears with the correct cyrillic, and it is playable as expected.

❌ **DRAG folder "[2000] Я сошла с ума"**, _or_
❌ **DRAG folder "t.A.T.u."**

For every file, a proper entry appears with whatever characters are in it, and is playable as expected.

For files with cyrillic in it (and only them!), a second entry also appears with all the cyrillic replaced by underscores, and playing it results in "Invalid" being shown in the playlist. See screenshot. At the bottom is the tooltip from hovering over the bottom-most entry.

![image](https://user-images.githubusercontent.com/399950/115807419-e2569f80-a3e8-11eb-8a21-9328c84edbf3.png)

_Note: these files are on a network drive. I don't think it has any bearing on this situation, but I want to mention it just in case._

# Expected behaviour

A single entry. The top one will do. 😄 

# Version information

MPC-HC (Nightly, 64-bit)
------------------------

Build information:
    Version:            1.9.10.57 (7496b042e)
    Compiler:           MSVC v19.26.28806
    Build date:         Mar 30 2021

LAV Filters:
    LAV Splitter:       0.75.0.0
    LAV Video:          0.75.0.0
    LAV Audio:          0.75.0.0
    FFmpeg compiler:    MinGW-w64 GCC 9.3.0

Operating system:
    Name:               Windows NT 10.0 (build 19042)
    Version:            10.0 (64-bit)

Hardware:
    CPU:                Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz
    GPU:                NVIDIA GeForce GTX 760 (driver version: 27.21.14.5671)

_Edit: I just realized my K-Lite wasn't fully updated despite updating it just the other week. So I updated it completely, and even got a new MPC-HC version out of it. Unfortunately, it is still not fixed. Including information about this version, too._

MPC-HC (64-bit)
---------------

Build information:
    Version:            1.9.11 (6096e5ea7)
    Compiler:           MSVC v19.26.28806
    Build date:         Apr 22 2021

LAV Filters:
    LAV Splitter:       0.75.0.2
    LAV Video:          0.75.0.2
    LAV Audio:          0.75.0.2
    FFmpeg compiler:    MinGW-w64 GCC 9.3.0

Operating system:
    Name:               Windows NT 10.0 (build 19042)
    Version:            10.0 (64-bit)

Hardware:
    CPU:                Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz
    GPU:                NVIDIA GeForce GTX 760 (driver version: 27.21.14.5671)

**Comment by clsid2** [comment]
I just tried with exact same folder and file names and it worked normally. No double entries.

I tested both Windows 7 and Windows 10 19041.746

**Comment by worstje** [comment]
Because you couldn't reproduce, I have copied the tree structure in question to a local drive in case the network drive caused the problem somehow. I have verified that it still occurs regardless of the physical storage location or underlying protocols used.

Is there any way I can help you further diagnose and identify this issue?

**Comment by clsid2** [comment]
I made a change that might help. Try next development build when it becomes available (no build planned yet atm).

**Comment by ale5000-git** [comment]
@worstje: Have you tried [1.9.11.14](https://github.com/clsid2/mpc-hc/releases/tag/1.9.11.14)?

**Comment by worstje** [comment]
@ale5000-git : I just tried it using the zip package that you have linked. Nothing has changed, unfortunately.

See accompanying screenshot which includes the About box for proof.

![image](https://user-images.githubusercontent.com/399950/116678748-fd597e80-a9a9-11eb-9579-d3fdf6912c5a.png)

Thank you for looking into this.

**Comment by clsid2** [comment]
It uses the FindFirstFileW/FindNextFileW Windows API for enumerating the files in the folder. So the problem must be in Windows itself. I can't reproduce with the same file and folder names.

**Comment by ale5000-git** [comment]
@worstje: Could you please try to create a folder with one or maximum 2 files and reproduce the problem? If yes, please post the files itself so I can try.

**Comment by worstje** [comment]
I've done some more testing, and it seems I have misrepresented the issue in my ignorance. After more in-depth testing I found out that MPC-HC apparently opens .cue files nowadays. (If it was always able to do this then I don't have a clue how I managed to avoid bumping into this feature during the past decade!)

In my brain the primary purpose of that file is accurately burning CDs, so I hadn't considered it as a part of the problem at all. I thought it would be ignored just like album art, ,accurip and other log files, because I never saw these files themselves pop up in the playlist.

To capitalize on my misunderstanding, it turns out that my other randomly-chosen test artist with boring latin characters didn't happen to contain a single .cue file, so I had honed in on the wrong suspect with one of my first tests that apparently didn't test what I thought it was testing!

My .CUE files tend to be horribly invalid  because I retag/rename my collection using more convenient tools, leading them to go out of sync. I only hang on to them in case I need them some day to reproduce a CD accurately, at which point I can just fix up whatever filenames exist.

I am so sorry for sending you both on a wild goose chase. I truly thought I did my due diligence! 🤦 God, I feel like an utter idiot right now.

Does a setting exist for MPC to ignore .CUE files? Or just skip adding entries that don't physically exist at least?
